### PR TITLE
Bug #38 fix:  While adding T&C either Global or URL pattern need to required otherwise user is able to T&C without selecting global and URL pattern

### PR DESCRIPTION
**Pull request by paritshivani**

### CODE REVIEW

File: src/components/com_tc/administrator/models/content.php [hunk 1]
```diff
@@ -175,6 +175,10 @@ protected function prepareTable($table)
 	 */
 	public function save($data)
 	{
+		if(!$this->validator($data))
```
Comment (snehal0904) [inline src/components/com_tc/administrator/models/content.php]: @paritshivani Call this from controller save method

File: src/components/com_tc/administrator/controllers/content.php [hunk 2]
```diff
@@ -61,4 +61,65 @@ public function checkDuplicateAndLatestVersionTC()
...
+		* Function to save field data
+		*
+		* @param   string  $key     key
+		* @param   string  $urlVar  urlVar
+		*
+		* @return  void
+		*/
+	public function save($key = null, $urlVar = null)
+	{
+		$app = JFactory::getApplication();
```
Comment (snehal0904) [inline src/components/com_tc/administrator/controllers/content.php]: @paritshivani Use namspace for new functionalities

File: src/components/com_tc/administrator/controllers/content.php [hunk 3]
```diff
@@ -61,4 +61,65 @@ public function checkDuplicateAndLatestVersionTC()
...
 		jexit();
 	}
+
+	/**
+		* Function to save field data
+		*
+		* @param   string  $key     key
+		* @param   string  $urlVar  urlVar
+		*
+		* @return  void
```
Comment (snehal0904) [inline src/components/com_tc/administrator/controllers/content.php]: @paritshivani return boolean|void

File: src/components/com_tc/administrator/controllers/content.php [hunk 4]
```diff
@@ -61,4 +64,65 @@ public function checkDuplicateAndLatestVersionTC()
...
+	/**
+		* Function to save field data
+		*
+		* @param   string  $key     key
+		* @param   string  $urlVar  urlVar
+		*
+		* @return  boolean|void
+		*/
+	public function save($key = null, $urlVar = null)
+	{
```
Comment (manojLondhe) [inline src/components/com_tc/administrator/controllers/content.php:77]: ensure this is similar to https://github.com/manojLondhe/joomla-extensions-generator/blob/master/templates/component/administrator/controllers/form.php

File: src/components/com_tc/administrator/models/content.php [hunk 5]
```diff
@@ -575,4 +576,34 @@ public function isUserAcceptedTC($loggedInUserId, $tcId)
...
+			$option = $pattern['option'];
+			$view = $pattern['view'];
+
+			if ($global == 0 && (empty($option) || empty($view)))
+			{
+				return false;
+			}
+		}
+
+				return parent::validate($form, $data, $group);
```
Comment (manojLondhe) [inline src/components/com_tc/administrator/models/content.php]: fix alignment?

**Pull request closed without merging**